### PR TITLE
open street map layer

### PR DESCRIPTION
**Pull request by markjdugger**
Demonstrate use of open street map as a basemap.

### CODE REVIEW

**Comment by markjdugger** [comment]
@tschie Do you accept design rationale--to use open street map as basemap in this sample?

**Comment by markjdugger** [comment]
I also got thoughtful responses on this issue via Slack thread:
https://esri-runtime.slack.com/archives/C0HPF1Q87/p1537192449000100?thread_ts=1536968748.000100&cid=C0HPF1Q87
Lucas Danzinger's "final" word convinced me:

> I think the most common use case for displaying a OSM Layer is to set the map/scene’s basemap to OSM, so I think it makes more sense to have it be in the basemap and not operational layer list

**Comment by markjdugger** [comment]
this is test
- [x] design runtime/common-samples#129

**Comment by markjdugger** [comment]
#129

**Comment by markjdugger** [comment]
[](url)

**Comment by markjdugger** [comment]
*https://devtopia.esri.com/runtime/common-samples/issues/129

**Comment by markjdugger** [comment]
@tschie Do you approve this PR?

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017 Esri.
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java]: need to change to 2018

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      StackPane stackPane = new StackPane();
+      Scene scene = new Scene(stackPane);
+
+      // set title, size, and add scene to stage
+      stage.setTitle("Open Street Map Sample");
+      stage.setWidth(800);
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java:46]: remove new line

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+      // create an ArcGISMap with an OPEN_STREET_MAP basemap
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java]: create ~~an ArcGISMap~~ map with an ~~OPEN_STREET_MAP~~ OpenStreetMap basemap

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      ArcGISMap map = new ArcGISMap(Basemap.Type.OPEN_STREET_MAP, 34.056295, -117.195800, 10);
+      
+      // set the map to be displayed in this view
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java]: "set the map to a map view"

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      mapView = new MapView();
+      mapView.setMap(map);
+	  	  
+      // add the map view to stack pane
+      stackPane.getChildren().addAll(mapView);
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java]: addAll -> add

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
+# OpenStreetMap Layer
+Add the OpenStreetMap layer to a map as a Basemap.
+
+![OpenStreetMap Layer App](openstreetmap-layer.png)
+
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md:4]: This file needs to be in html while still having the .md extension. The reason is the sample viewer renders these in a webview as html. Look at the raw text of some of the other sample readmes to get an idea of the format.

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
+# OpenStreetMap Layer
+Add the OpenStreetMap layer to a map as a Basemap.
+
+![OpenStreetMap Layer App](openstreetmap-layer.png)
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md]: The image needs to be 800w by 400h.

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md [hunk 8]
```diff
@@ -0,0 +1,18 @@
+# OpenStreetMap Layer
+Add the OpenStreetMap layer to a map as a Basemap.
+
+![OpenStreetMap Layer App](openstreetmap-layer.png)
+
+## How to use the sample
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md]: You can just omit this section for basic samples like this one.

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
+# OpenStreetMap Layer
+Add the OpenStreetMap layer to a map as a Basemap.
+
+![OpenStreetMap Layer App](openstreetmap-layer.png)
+
+## How to use the sample
+Simply run the app.
+
+## How it works
+Instantiate an `ArcGISMap` with a `Basemap.Type.OPEN_STREET_MAP` and add it to a `MapView`. The attribution text that is required by OpenStreetMap is automatically added to the `MapView's` attributionText without any additional code.
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md]: This is always shown as a list for Java SE. Make this a two step list with the first step to create the map and the second to set it to the map view. The part about the attribution can be omitted.

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+![OpenStreetMap Layer App](openstreetmap-layer.png)
+
+## How to use the sample
+Simply run the app.
+
+## How it works
+Instantiate an `ArcGISMap` with a `Basemap.Type.OPEN_STREET_MAP` and add it to a `MapView`. The attribution text that is required by OpenStreetMap is automatically added to the `MapView's` attributionText without any additional code.
+
+## Relevant API
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md]: Add `Basemap.Type`

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+* ArcGISMap
+* Basemap
+* MapView
+
+#### Tags
+Layers
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md]: Add OpenStreetMap

**Review by tschie — CHANGES_REQUESTED** [review]
Great first start! I've suggested a few changes based on the sample implementation guidelines (I sent you a link separately). Please re-request review after you've made the changes.

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md [hunk 12]
```diff
@@ -1,18 +1,26 @@
...
-Add the OpenStreetMap layer to a map as a Basemap.
+<h1>OpenStreetMap Layer</h1>
 
-![OpenStreetMap Layer App](openstreetmap-layer.png)
+<p>Add the OpenStreetMap layer to a map as a Basemap.</p>
 
-## How to use the sample
-Simply run the app.
+<img src="OpenStreetMapLayer.png"/>
+<figcaption>OpenStreetMap Layer App</figcaption>
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md]: Don't need the `figcaption` or `figure`. I also don't think the renderer in the sample viewer will handle it.

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java [hunk 13]
```diff
@@ -44,7 +44,7 @@ public void start(Stage stage) {
       stage.setScene(scene);
       stage.show();
 
-      // create an ArcGISMap with an OPEN_STREET_MAP basemap
+      // create a map with an OpenSteetMap basemap
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java]: typo Steet

File: src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+<h2>Relevant API</h2>
+
+<ul>
+   <li>ArcGISMap</li>
+   <li>Basemap</li>
+   <li>Basemap.Type</li>
+   <li>MapView</li>
+</ul>
+
+<h4>Tags</h4>
```
Comment (tschie) [inline src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md]: use h2 tag here

**Review by tschie — APPROVED** [review]
Looks good! Only one more thing to change (the `h2` tag in the readme). Feel free to merge after that.

**Pull request merged**